libats-auth/src/main/scala/com/advancedtelematic/libats/auth/TokenValidator.scala

```diff
@@ -0,0 +1,127 @@
+package com.advancedtelematic.libats.auth
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.client.RequestBuilding._
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.ContentTypes._
+import akka.http.scaladsl.model.{HttpEntity, Multipart, StatusCodes, Uri}
+import akka.http.scaladsl.model.headers.{Authorization, BasicHttpCredentials, OAuth2BearerToken}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.server.{AuthorizationFailedRejection, Directive0, Directives}
+import akka.http.scaladsl.util.FastFuture
+import akka.stream.ActorMaterializer
+import com.advancedtelematic.akka.http.jwt.JwtDirectives
+import com.advancedtelematic.json.signature.JcaSupport
+import com.advancedtelematic.jwa.HS256
+import com.advancedtelematic.jws.Jws
+import com.typesafe.config.{ConfigException, ConfigFactory}
+import io.circe.Json
+import io.circe.generic.auto._
+import javax.crypto.SecretKey
+import javax.crypto.spec.SecretKeySpec
+import cats.syntax.either._
+import org.apache.commons.codec.binary.Base64
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+case class ValidationResponse(active: Boolean)
+
+object TokenValidator {
+  def apply()(implicit system: ActorSystem, mat: ActorMaterializer) = new TokenValidator
+}
+
+class TokenValidator(implicit system: ActorSystem, mat: ActorMaterializer) {
```

Comment:
In Treehub there is a comment that you want this to be a `Materializer` instead of `ActorMaterializer`